libbpf-tools/softirqs.bpf.c

```diff
@@ -32,11 +43,14 @@ static int handle_entry(unsigned int vec_nr)
 
 static int handle_exit(unsigned int vec_nr)
 {
+        if (!is_target_cpu())
+	        return 0;
+
 	u64 delta, *tsp;
 	u32 key = 0;
 
-	if (vec_nr >= NR_SOFTIRQS)
-		return 0;
+	vec_nr &= 0xf;  // to avoid compiler opti make verifier error
```

Comment:
which compiler do you use and what kind of error you hit on which kernel?

Reply:
Please give details about the above verifier failure (kernel version, compiler version, error message, etc.)

Reply:
see https://github.com/iovisor/bcc/pull/5107#issuecomment-2383892592

Reply:
Sorry, I missed the original comments.
From the verifier log, the following is the reason:
```
18: (bf) r1 = r6                      ; frame1: R1_w=scalar(id=1) R6=scalar(id=1)
19: (67) r1 <<= 32                    ; frame1: R1_w=scalar(smax=0x7fffffff00000000,umax=0xffffffff00000000,smin32=0,smax32=umax32=0,var_off=(0x0; 0xffffffff00000000))
20: (77) r1 >>= 32                    ; frame1: R1_w=scalar(smin=0,smax=umax=0xffffffff,var_off=(0x0; 0xffffffff))
21: (25) if r1 > 0x9 goto pc+72       ; frame1: R1_w=scalar(smin=smin32=0,smax=umax=smax32=umax32=9,var_off=(0x0; 0xf))
22: (bf) r2 = r10                     ; frame1: R2_w=fp0 R10=fp0
...
39: (67) r6 <<= 32                    ; frame1: R6_w=scalar(smax=0x7fffffff00000000,umax=0xffffffff00000000,smin32=0,smax32=umax32=0,var_off=(0x0; 0xffffffff00000000))
40: (77) r6 >>= 32                    ; frame1: R6_w=scalar(smin=0,smax=umax=0xffffffff,var_off=(0x0; 0xffffffff))
...
```
In the above, at insn 22, we know r1 <= 0x9, and not that insn 18, we have r1 = r6, so lower 32bit of r6 should be <= 0x9 as well.
After insn 40, we know r6 top 32bits are 0, so in ideal case verifier should conclude r6 <= 0x9, but unfortunately verifier is not sophisticated enough to derive the precise bound.

I suspect -mcpu=v3 might work as we could remove some left/right shifts.

Reply:
Great. `-mcpu=v3` works well. 

I added 
```
BPFCFLAGS_softirqs := $(BPFCFLAGS) -mcpu=v3
$(OUTPUT)/softirqs.bpf.o: BPFCFLAGS = $(BPFCFLAGS_softirqs)
```

I wonder if there is a better way to add `-mcpu=v3` for softirqs. If so, I will modify and rebase the commit.